Added additional settings.

GuidoZam opened the pull request:
## 🎯 Aim

Added a new additional step setting when scaffolding a new project to specify if the node version manager configuration file should be created or not.

## 📷 Result

![image](https://github.com/user-attachments/assets/e99d1ee8-2c0a-4070-8252-40086a50b222)

![image](https://github.com/user-attachments/assets/bca41a6a-7247-49e0-a51c-adb77d86982b)

## ✅ What was done

- [X] Added a new project additional step to create or not the node version manager configuration file.
- [X] Added new VSCode setting `createNodeVersionFileDefaultValue` to define the default value of the new project additional step.
- [X] Added new VSCode setting `nodeVersionManagerFile` to specify which configuration file to be created.

## 🔗 Related issue

Closes: #285

Adam-it commented:
Awesome 🤩😍. I will review this ASAP

Adam-it commented on src/services/actions/Scaffolder.ts:
```diff
@@ -147,6 +147,45 @@ export class Scaffolder {
+	 * Returns the value of the nodeVersionManager setting and sends it to the webview.
+	 */
+	public static async nodeVersionManager() {
+		const value = getExtensionSettings<string>('nodeVersionManager', false);
```
`false` should not be a default value of a setting which we rather expect to return a string. I also introduced a bug fix connected to that just a while ago which will most probably make this not working in order to ensure correct types. 
lets update  this to `nvm` as default
```suggestion
		const value = getExtensionSettings<string>('nodeVersionManager', 'nvm');
```

Adam-it commented on src/services/actions/Scaffolder.ts:
```diff
@@ -147,6 +147,45 @@ export class Scaffolder {
+	 * Returns the value of the nodeVersionManagerFile setting and sends it to the webview.
+	 */
+	public static async nodeVersionManagerFile() {
+		const value = getExtensionSettings<string>('nodeVersionManagerFile', false);
```
also here the default value should be string. Since in `package.json` we specify the default is `.nvmrc` lets do the same here

Adam-it commented on src/services/actions/Scaffolder.ts:
```diff
@@ -225,6 +264,29 @@ export class Scaffolder {
+                break;
+                case 'none':
+                // If the node version manager is none, do not create any file
+                break;
+							}
+						}
+
```
I just must ask since GH is rendering the whitespace very strange in this PR. 
what is encoding and set and space indentation in VS Code?
![image](https://github.com/user-attachments/assets/222f7fdd-209c-4940-b21a-ecbe03a7488f)

GuidoZam replied:
It's strange but in some way VS Code has added tabs instead of spaces to indent, I'll fix this as well.

Adam-it commented on src/services/actions/Scaffolder.ts:
```diff
@@ -225,6 +264,29 @@ export class Scaffolder {
+                    break;
+                }
+                break;
+                case 'none':
```
perfect case to be default as well. Lets add this case as `default` so that if in future we will add some new node version manager by default we will not create any file 

```suggestion
                case 'none':
                default:
```

Adam-it commented on src/constants/WebviewCommand.ts:
```diff
@@ -17,5 +22,10 @@ export const WebviewCommand = {
+		createNodeVersionFileDefaultValue: 'should-create-node-version-file',
+		nodeVersionManager: 'node-version-manager',
+		nodeVersionManagerFile: 'node-version-manager-file',
+		createNVMRCFile: 'create-nvmrc-file',
+		createNodeVersionManagerFile: 'node-version-manager-file',
```
seems those two are never sent from webview to vscode. Lets remove them

```suggestion
```

Adam-it commented on src/constants/WebviewCommand.ts:
```diff
@@ -5,6 +5,11 @@ export const WebviewCommand = {
+		createNodeVersionFileDefaultValue: 'should-create-node-version-file',
+		nodeVersionManager: 'node-version-manager',
+		nodeVersionManagerFile: 'node-version-manager-file',
+		createNVMRCFile: 'create-nvmrc-file',
```
is this used? I am almost sure this one is not used and we may remove it. We use a similar constant in `ProjectFileContent.ts`

Adam-it reviewed (changes requested):
@GuidoZam Awesome work so far 👏👏👏👏👏. You Rock 🤩
I left a few comments. Please do give them a recheck.
When testing I also found some 2 strange behaviors we should fixup before we merge:

I don't know why but when I select creating nvm file together with other dependencies 
![image](https://github.com/user-attachments/assets/516cbaf2-b92b-4653-a8d1-3c6aa2ac4e2c)
it is set in the project.pnp file 
![image](https://github.com/user-attachments/assets/2dbecf4e-a5fb-420a-bf60-b86d726ec71c)
but as a result in terminal I only get the nvm file installed no other npm install is running 
![image](https://github.com/user-attachments/assets/444937ee-1193-463c-8ea2-dd6c2817e370)
I don't know why but this seem to be only like this when I have this new dependency set otherwise it works normally and installed everything so it seems like a bug

also 
when you reset the create nvm file to default it will be set to true not false by default
![image](https://github.com/user-attachments/assets/d04d3c16-2cbc-4208-94d3-ae4aabaf82b6)

most probably this is due to the fact that in `package.json` we set this a string "false" not `false`

Adam-it commented:
@GuidoZam I also forgot to mention that we should update the readme part with additional step in scaffolding form to now include this new option as well together with screenshots

Adam-it commented on data/sp-dev-fx-samples.json:
@GuidoZam just a quick heads up. This file should not be modified during a PR . 
in the repo there is a separate GH Action bot that modifies this one

GuidoZam commented:
Hi @Adam-it, I verified and fixed all your reviews.
The two bugs are resolved:
- The first one I think was a different behavior produced by the custom name of the terminal. I've removed that customization and everything's seems to work fine.
- About the second bug you were right, it was just a type mistake, updating the `package.json` type resolved the issue.

I've also updated the README file, have a look and tell me if this way is ok for you. There are some new images and text for the Node version manager support (section 12 of the README).

About the `data/sp-dev-fx-samples.json` file I didn't update that on purpose, I guess I've missed something while mergin' the branches.

Let me know if there's something else to update (hopefully not 🤞).

Adam-it reviewed (approved):
LGTM  👍 and works perfectly 👏
I did a couple of small fixups
